dev/doc/critical-bugs

```diff
@@ -0,0 +1,221 @@
+  introduced: 10 Sep 2014, 6624459e4, Sozeau (?)
+  impacted released versions: V8.5
+  impacted development branches: none
+  impacted coqchk versions: ?
+  fixed in: trunk/master/v8.5 (120053a50, 4 Mar 2016, Dénès)
+  found by: Dénès exploiting bug #4588
+  exploit: test-suite/bugs/closed/4588.v
+  GH issue number: #4588
+  risk: ?
+
+Conversion machines
+
+  component: "virtual machine" (compilation to bytecode ran by a C-interpreter)
+  summary: collision between constructors when more than 256 constructors in a type
+  introduced: V8.1
+  impacted released versions: V8.1-V8.5pl3, V8.2-V8.2pl2, V8.3-V8.3pl3, V8.4-V8.4pl5
+  impacted development branches: none
+  impacted coqchk versions: none (no virtual machine in coqchk)
+  fixed in: master/trunk/v8.5 (00894adf6/596a4a52, 26-39 Mar 2015, Grégoire), v8.4 (cd2101a39, 1 Apr 2015, Grégoire), v8.3 (a0c7fc05b, 1 Apr 2015, Grégoire), v8.2 (2c6189f61b8, 1 Apr 2015, Grégoire), v8.1 (bb877e5b, 29 Nov 2015, Herbelin, backport)
+  found by: Dénès, Pédrot
+  exploit: test-suite/failure/vm-bug4157.v
+  GH issue number: #4157
+  risk: 
+
+  component: "virtual machine" (compilation to bytecode ran by a C-interpreter)
+  summary: wrong universe constraints
+  introduced: possibly exploitable from V8.1; exploitable at least from V8.5
+  impacted released versions: V8.1-V8.4pl5 unknown, V8.5-V8.5pl3, V8.6-V8.6.1, V8.7.0-V8.7.1
+  impacted development branches: unknown for v8.1-v8.4, none from v8.5
+  impacted coqchk versions: none (no virtual machine in coqchk)
+  fixed in: master (c9f3a6cbe5c41, 12 Feb 2018, PR#6713, Dénès), v8.7 (c058a4182, 15 Feb 2018, Zimmermann, backport), v8.6 (a2cc54c6, 21 Feb 2018, Herbelin, backport), v8.5 (d4d550d0f1a, 21 Feb 2018, Herbelin, backport)
+  found by: Dénès
+  exploit: test-suite/bugs/closed/6677.v
+  GH issue number: #6677
+  risk: 
+
+  component: "virtual machine" (compilation to bytecode ran by a C-interpreter)
+  summary: missing pops in executing 31bit arithmetic
+  introduced: V8.5
+  impacted released versions: V8.1-V8.4pl5
+  impacted development branches: v8.1 (probably)
+  impacted coqchk versions: none (no virtual machine in coqchk)
+  fixed in: master/trunk/v8.5 (a5e04d9dd, 6 Sep 2015, Dénès), v8.4 (d5aa3bf6, 9 Sep 2015, Dénès), v8.3 (5da5d751, 9 Sep 2015, Dénès), v8.2 (369e82d2, 9 Sep 2015, Dénès), 
+  found by: Catalin Hritcu
+  exploit: lost?
+  GH issue number: ?
+  risk: 
+
+  component: "native" conversion machine (translation to OCaml which compiles to native code)
+  summary: translation of identifier from Coq to OCaml was not bijective, leading to identify True and False
+  introduced: V8.5
+  impacted released versions: V8.5-V8.5pl1
+  impacted development branches: none
+  impacted coqchk versions: none (no native computation in coqchk)
+  fixed in: master/trunk/v8.6 (a91d15dcb4, 19 May 2016, letouzey), v8.5 (088b3161c93, 19 May 2016, letouzey), 
+  found by: Letouzey, Dénès
+  exploit: lost?
```

Comment:
I think this one might actually be 244d7a9aafe7ad613dd2095ca3126560cb3ea1d0, which includes the exploit in the commit message?

> `Unicode.ascii_of_ident` is now truly injective
>
> A non-ASCII char is now converted to `_UUxxxx_` with xxxx being its unicode index
 in hexa. And any preexisting `_UU` substring in the ident is converted to `_UUU`.
 The switch from `__Uxxxx_` to `_UUxxxx_` is cosmetic, it just helps the extraction
 (less __ in names). But the other part of the patch (detection of preexisting
 `_UU` substrings) is critical to make `ascii_of_ident` truly injective and avoid
 the following kind of proof of False via native_compute :
>
>      Definition α := 1.
>      Definition __U03b1_ := 2.
>      Lemma oups : False.
>      Proof.
>       assert (α = __U03b1_). { native_compute. reflexivity. }
>       discriminate.
>      Qed.

I think the commit you're pointing at (a91d15dcb4) is just cosmetic?

Reply:
Right. Fixed. Thanks.